### PR TITLE
copy environment service and it's dependencies

### PR DESCRIPTION
**Pull request by biharitomi**

### CODE REVIEW

File: environment/src/main/java/com/sequenceiq/environment/environment/converter/EnvironmentViewToSimpleEnvironmentV1ResponseConverter.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        return response;
+    }
+
+    private void setNetworkIfPossible(SimpleEnvironmentV1Response response, EnvironmentView source) {
+        BaseNetwork network = source.getNetwork();
+        if (network != null) {
+            CloudPlatform cloudPlatform = CloudPlatform.valueOf(source.getCloudPlatform());
+            EnvironmentNetworkConverter environmentNetworkConverter = environmentNetworkConverterMap.get(cloudPlatform);
+            if (environmentNetworkConverter != null) {
+                EnvironmentNetworkV1Response networkV4Response = environmentNetworkConverter.convert(source.getNetwork());
```
Comment (afarsang) [inline environment/src/main/java/com/sequenceiq/environment/environment/converter/EnvironmentViewToSimpleEnvironmentV1ResponseConverter.java]: EnvironmentNetworkV1Response networkResponse = .. would be nice

File: environment/src/main/java/com/sequenceiq/environment/environment/converter/network/AwsEnvironmentNetworkConverter.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    BaseNetwork createProviderSpecificNetwork(EnvironmentNetworkV1Request source) {
+        AwsNetwork awsNetwork = new AwsNetwork();
+        awsNetwork.setVpcId(source.getAws().getVpcId());
+        return awsNetwork;
+    }
+
+    @Override
+    EnvironmentNetworkV1Response setProviderSpecificFields(EnvironmentNetworkV1Response result, BaseNetwork network) {
+        AwsNetwork awsNetwork = (AwsNetwork) network;
+        EnvironmentNetworkAwsV1Params awsV4Params = new EnvironmentNetworkAwsV1Params();
```
Comment (afarsang) [inline environment/src/main/java/com/sequenceiq/environment/environment/converter/network/AwsEnvironmentNetworkConverter.java]: EnvironmentNetworkAwsV1Params awsV1Params =..

File: environment/src/main/java/com/sequenceiq/environment/environment/converter/network/AzureEnvironmentNetworkConverter.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        azureNetwork.setResourceGroupName(azureParams.getResourceGroupName());
+        azureNetwork.setNoPublicIp(azureParams.getNoPublicIp());
+        azureNetwork.setNoFirewallRules(azureParams.getNoFirewallRules());
+        return azureNetwork;
+    }
+
+    @Override
+    EnvironmentNetworkV1Response setProviderSpecificFields(EnvironmentNetworkV1Response result, BaseNetwork network) {
+        AzureNetwork azureNetwork = (AzureNetwork) network;
+        EnvironmentNetworkAzureV1Params azureV4Params = new EnvironmentNetworkAzureV1Params();
```
Comment (afarsang) [inline environment/src/main/java/com/sequenceiq/environment/environment/converter/network/AzureEnvironmentNetworkConverter.java]: EnvironmentNetworkAzureV1Params azureV1Params ...

File: environment/src/main/java/com/sequenceiq/environment/environment/validator/network/AzureEnvironmentNetworkValidator.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+import com.sequenceiq.cloudbreak.util.ValidationResult;
+import com.sequenceiq.environment.CloudPlatform;
+import com.sequenceiq.environment.api.environment.model.EnvironmentNetworkAzureV1Params;
+import com.sequenceiq.environment.api.environment.model.request.EnvironmentNetworkV1Request;
+
+@Component
+public class AzureEnvironmentNetworkValidator implements EnvironmentNetworkValidator {
+
+    @Override
+    public void validate(EnvironmentNetworkV1Request networkV4Request, ValidationResult.ValidationResultBuilder resultBuilder) {
```
Comment (afarsang) [inline environment/src/main/java/com/sequenceiq/environment/environment/validator/network/AzureEnvironmentNetworkValidator.java]: public void validate(EnvironmentNetworkV1Request networkV1Request, ValidationResult.ValidationResultBuilder resultBuilder)

File: environment/src/main/java/com/sequenceiq/environment/environment/validator/network/EnvironmentNetworkValidator.java [hunk 5]
```diff
@@ -0,0 +1,21 @@
+package com.sequenceiq.environment.environment.validator.network;
+
+
+import com.sequenceiq.cloudbreak.util.ValidationResult;
+import com.sequenceiq.environment.CloudPlatform;
+import com.sequenceiq.environment.api.environment.model.request.EnvironmentNetworkV1Request;
+
+public interface EnvironmentNetworkValidator {
+
+    void validate(EnvironmentNetworkV1Request networkV4Request, ValidationResult.ValidationResultBuilder resultBuilder);
```
Comment (afarsang) [inline environment/src/main/java/com/sequenceiq/environment/environment/validator/network/EnvironmentNetworkValidator.java]: void validate(EnvironmentNetworkV1Request networkV1Request, ValidationResult.ValidationResultBuilder resultBuilder);

File: environment/src/main/java/com/sequenceiq/environment/credential/Credential.java [hunk 6]
```diff
@@ -19,7 +21,7 @@
 
 @Entity
 @Table(uniqueConstraints = @UniqueConstraint(columnNames = {"workspace_id", "name"}))
-public class Credential implements WorkspaceAwareResource {
+public class Credential implements WorkspaceAwareResource, Serializable {
```
Comment (topolyai5) [inline environment/src/main/java/com/sequenceiq/environment/credential/Credential.java]: is serializable important?

Comment (biharitomi) [reply environment/src/main/java/com/sequenceiq/environment/credential/Credential.java]: yes, it is a spot bug discovered issue

**Pull request merged**